cmake/functions.cmake

```diff
@@ -45,8 +45,8 @@ function(add_flag flag)
 endfunction()
 
 function(compile_proto_to_cpp PROTO_LIBRARY_NAME PB_H PB_CC PROTO)
-  get_target_property(Protobuf_INCLUDE_DIR protobuf::libprotobuf INTERFACE_INCLUDE_DIRECTORIES)
-  get_target_property(Protobuf_PROTOC_EXECUTABLE protobuf::protoc IMPORTED_LOCATION_RELEASE)
+  #get_target_property(Protobuf_INCLUDE_DIR protobuf::libprotobuf INTERFACE_INCLUDE_DIRECTORIES)
```

Comment:
Right, I don't intend to leave commented code - commit is marked [WIP]. They should probably be conditionally enabled, but on what condition? They broke my cross compile attempt, so maybe we switch on cross compiling? I don't think we should expose a CMake option.

Do you know why these lines were added in the first place?

Reply:
What I see here - in these two lines we are reading `INTERFACE_INCLUDE_DIRECTORIES` and `IMPORTED_LOCATION_RELEASE` parameters, which have to be set by `find_package(Protobuf ...)` command.

The error from the commit message says - it was not succeeded to find a proto compiler in your environment. Probably you can additionally specify own custom CMake toolchain file which will `set_target_property(...)`. Otherwise, it looks like proto couldn't be compiled without a compiler :)

We could employ [CMAKE_CROSSCOMPILING](https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html) var:
```cmake
if (NOT CMAKE_CROSSCOMPILING)
  get_target_property(Protobuf_INCLUDE_DIR protobuf::libprotobuf INTERFACE_INCLUDE_DIRECTORIES)
  get_target_property(Protobuf_PROTOC_EXECUTABLE protobuf::protoc IMPORTED_LOCATION_RELEASE)
endif()
``` 
BUT, the variables still have to be defined in another way.

Reply:
I tackled this the following way:

* [CMake: Allow build system to provide protobuf compiler](https://github.com/eigendude/libp2p/commit/cross-compile~3)